<a href="https://colab.research.google.com/github/mwcoyne2/BDI-Final-Project/blob/main/BDI_Final_Project_Final_Draft.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#College Basketball Conference Analysis


**Why I chose this data Set:**
I choose to work with the college basketball data set becuase I am a huge colelge basketball fan. Especially this year with Illinois doing well I watched every singly Illini game. I am the president of the student sections this year and devoted my entire college experience to sports. I thought it would be fun to perform data analytics on the 20-21 season expecially since I followed the season so closely. I also want to see if Illinois was actually that good statistically compared to other schools.

**Question for Analysis:**
The question I want to answer is if the big ten is better then the other conferences. I plan on doing this by comparing multiple stats across multiple conferences, specifically tournement seeding data. I will also analyze Offense and defense efficiency to see if there is a correlation to wins. If I have time I may also compare teams within the BIG 10

#Instalation and renaming Dataset
In this section Intalled the correct functions and renamed the data set for a better user expereience.

In [1]:
#Instalation Steps
if 'google.colab' in str(get_ipython()):
  !pip install plotly==4.14.3

import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio

In [2]:
#Importing Data Set
df = pd.read_csv('https://github.com/mwcoyne2/BDI-Final-Project/raw/main/2021%20College%20Basketball.csv')
display(df)

TEAM  CONF   G   W  ADJOE  ...  3P_O  3P_D  ADJ_T  WAB  SEED
0            Michigan   B10  24  20  118.1  ...  38.7  33.5   66.9  7.2   1.0
1              Baylor   B12  24  22  123.2  ...  41.8  34.0   68.8  6.6   1.0
2            Illinois   B10  29  23  117.7  ...  37.6  32.7   70.7  8.9   1.0
3             Gonzaga   WCC  26  26  125.4  ...  36.5  32.5   74.6  8.5   1.0
4                Iowa   B10  29  21  123.5  ...  38.6  34.8   70.0  5.6   2.0
..                ...   ...  ..  ..    ...  ...   ...   ...    ...  ...   ...
342    Louisiana Tech  CUSA  27  21  102.7  ...  34.6  29.6   69.6 -1.7   NaN
343            Toledo   MAC  29  21  113.3  ...  37.7  29.5   69.1 -2.1   NaN
344               UAB  CUSA  27  22  102.5  ...  31.1  32.4   67.5 -2.7   NaN
345  Eastern Kentucky   OVC  27  22  101.5  ...  35.8  33.0   75.1 -3.8   NaN
346           Belmont   OVC  29  26  108.5  ...  34.7  34.0   71.1 -1.1   NaN

[347 rows x 22 columns]

In [3]:
#I renamed columns so they are more user friendly for non basktball fans
df.rename(columns =
    {'TEAM':'Team',
    'CONF':'Conference',
    'G':'Games',
    'W':'Wins',
    'ADJOE':'Adj Offensive Efficiency',
    'ADJDE':'Adj Defensive Efficiency',
    'BARTHAG':'Power Ranking',
    'EFG_O':'Effective Field Goal Percentage Shot',
    'EFG_D':'Effective Field Goal Percentage Allowed',
    'TOR':'Turnover Rate',
    'TORD':'Steal Rate',
    'ORB':'Offensive Rebounds',
    'DRB':'Defensive Rebounds',
    'FTR':'Freee Throw Rate',
    'FTRD':'Free Throw Rate Allowed',
    '2P_O':'Two Point Percentage',
    '2P_D':'Two Point Percentage Allowed',
    '3P_O':'Three Point Percentage',
    '3P_D':'Three Point Percentage Allowed',
    'ADJ_T':'Adjusted Tempo',
    'WAB':'Wins Above Bubble',
    'POSTSEASON':'Postseason',
    'SEED':'Seed'},
    inplace = True
    )
df

Team Conference  ...  Wins Above Bubble  Seed
0            Michigan        B10  ...                7.2   1.0
1              Baylor        B12  ...                6.6   1.0
2            Illinois        B10  ...                8.9   1.0
3             Gonzaga        WCC  ...                8.5   1.0
4                Iowa        B10  ...                5.6   2.0
..                ...        ...  ...                ...   ...
342    Louisiana Tech       CUSA  ...               -1.7   NaN
343            Toledo        MAC  ...               -2.1   NaN
344               UAB       CUSA  ...               -2.7   NaN
345  Eastern Kentucky        OVC  ...               -3.8   NaN
346           Belmont        OVC  ...               -1.1   NaN

[347 rows x 22 columns]

#General Data Information
In this section I conducted general data analysis to understand what the data I am working with looks like

In [4]:
#Shape of Data Set
df.shape

(347, 22)

In [5]:
#Information on the data set
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 347 entries, 0 to 346
Data columns (total 22 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   Team                                     347 non-null    object 
 1   Conference                               347 non-null    object 
 2   Games                                    347 non-null    int64  
 3   Wins                                     347 non-null    int64  
 4   Adj Offensive Efficiency                 347 non-null    float64
 5   Adj Defensive Efficiency                 347 non-null    float64
 6   Power Ranking                            347 non-null    float64
 7   Effective Field Goal Percentage Shot     347 non-null    float64
 8   Effective Field Goal Percentage Allowed  347 non-null    float64
 9   Turnover Rate                            347 non-null    float64
 10  Steal Rate                               347 non-n

In [6]:
#Sample of Data set
display(df.head())
display(df.tail())
display(df.sample())

Team Conference  Games  ...  Adjusted Tempo  Wins Above Bubble  Seed
0  Michigan        B10     24  ...            66.9                7.2   1.0
1    Baylor        B12     24  ...            68.8                6.6   1.0
2  Illinois        B10     29  ...            70.7                8.9   1.0
3   Gonzaga        WCC     26  ...            74.6                8.5   1.0
4      Iowa        B10     29  ...            70.0                5.6   2.0

[5 rows x 22 columns]

Team Conference  ...  Wins Above Bubble  Seed
342    Louisiana Tech       CUSA  ...               -1.7   NaN
343            Toledo        MAC  ...               -2.1   NaN
344               UAB       CUSA  ...               -2.7   NaN
345  Eastern Kentucky        OVC  ...               -3.8   NaN
346           Belmont        OVC  ...               -1.1   NaN

[5 rows x 22 columns]

Team Conference  Games  ...  Adjusted Tempo  Wins Above Bubble  Seed
115  Alcorn St.       SWAC     19  ...            69.6               -9.5   NaN

[1 rows x 22 columns]

#General Distribution Graphs
In this section I did general distrubtion graphs. I decided to use wins as my distribution critera because they is a unverisal stat among teams and is typically determined by other stats

In [7]:
df.head()

Team Conference  Games  ...  Adjusted Tempo  Wins Above Bubble  Seed
0  Michigan        B10     24  ...            66.9                7.2   1.0
1    Baylor        B12     24  ...            68.8                6.6   1.0
2  Illinois        B10     29  ...            70.7                8.9   1.0
3   Gonzaga        WCC     26  ...            74.6                8.5   1.0
4      Iowa        B10     29  ...            70.0                5.6   2.0

[5 rows x 22 columns]

In [8]:
fig = px.histogram(
    df,
    x = 'Wins',
    title = 'Histogram of Wins'
)
fig.show()

In [9]:
fig = px.density_heatmap(
    df,
    x = 'Conference',
    y = 'Wins',
    title = 'Heatmap of Conference Wins'
)
fig.show()

**Findings:** After analyzing the Distributions of wins with both a histogram and a heat map it appears that the average team won 12-13 games with a slight skew toward less wins. Also after looking at the heat map it appears that the Big 10, Big 12, PAC 12, and Conference USA had the most wins as they were the only conferences with 3+ teams with 20+ wins

#Offensive Statistics Analysis
**What I did:** The first thing I wanted to analyze is the effect Offens has on wininng. To to this I graphed the offense efficiency to wins as well as graphed a bar chart of three point percentage. I took the average of each of these metrics to make it easier and compare conferences better.

In [10]:
df.head()

Team Conference  Games  ...  Adjusted Tempo  Wins Above Bubble  Seed
0  Michigan        B10     24  ...            66.9                7.2   1.0
1    Baylor        B12     24  ...            68.8                6.6   1.0
2  Illinois        B10     29  ...            70.7                8.9   1.0
3   Gonzaga        WCC     26  ...            74.6                8.5   1.0
4      Iowa        B10     29  ...            70.0                5.6   2.0

[5 rows x 22 columns]

In [11]:
df_average_offensive_stats = df.groupby('Conference', as_index=False).agg({
    'Wins':'mean',
    'Adj Offensive Efficiency':'mean',
    'Effective Field Goal Percentage Shot':'mean',
    'Offensive Rebounds':'mean',
    'Two Point Percentage':'mean',
    'Three Point Percentage':'mean'})
df_average_offensive_stats = df_average_offensive_stats.sort_values('Wins', ascending = False)
df_average_offensive_stats

Conference       Wins  ...  Two Point Percentage  Three Point Percentage
5         B10  15.642857  ...             50.085714               34.378571
6         B12  15.400000  ...             50.070000               34.520000
21        P12  15.000000  ...             49.916667               34.925000
25        SEC  14.428571  ...             50.471429               33.071429
24         SC  14.400000  ...             52.990000               33.860000
17        MVC  14.100000  ...             51.680000               34.330000
12       CUSA  14.000000  ...             49.814286               33.550000
23         SB  13.916667  ...             47.833333               34.066667
18        MWC  13.545455  ...             50.281818               33.754545
20        OVC  13.500000  ...             50.916667               33.550000
7          BE  13.454545  ...             50.209091               33.563636
1         ACC  13.066667  ...             50.853333               34.573333
30        WCC  12.600000  ...             51.200000               32.730000
3        ASun  12.111111  ...             50.888889               33.977778
4        Amer  12.090909  ...             47.336364               32.918182
8        BSky  11.727273  ...             49.472727               34.390909
15        MAC  11.666667  ...             49.975000               33.816667
27       Slnd  11.461538  ...             49.200000               34.115385
13       Horz  11.416667  ...             50.908333               34.391667
9        BSth  11.181818  ...             50.318182               33.418182
10         BW  11.090909  ...             48.790909               34.718182
0         A10  10.714286  ...             50.700000               32.985714
28        Sum  10.555556  ...             49.011111               35.755556
29        WAC  10.222222  ...             48.177778               32.144444
14       MAAC  10.000000  ...             47.118182               32.372727
19        NEC   9.600000  ...             49.610000               33.060000
11        CAA   9.200000  ...             49.390000               33.320000
2          AE   8.900000  ...             48.580000               32.560000
26       SWAC   8.600000  ...             45.180000               31.200000
22        Pat   8.100000  ...             51.190000               34.210000
16       MEAC   7.555556  ...             47.133333               32.566667

[31 rows x 7 columns]

In [12]:
fig = px.bar(
    df_average_offensive_stats,
    x = 'Conference',
    y ='Adj Offensive Efficiency',
    title = 'Offensive Efficiency by Conference',
    color = 'Conference'
)
fig.show()

In [13]:
px.scatter(
    df_average_offensive_stats,
    x = 'Adj Offensive Efficiency',
    y = 'Wins',
    title = 'Is there a correlatuon between Two point and Three Point Percentage?',
    text = 'Conference',
    color = 'Conference'
)

In [14]:
fig = px.bar(
    df_average_offensive_stats,
    x = 'Conference',
    y = 'Three Point Percentage',
    title = 'Conference With The Best Three Point Percentage',
    color = 'Conference',
    
)
fig.show()

**Findings:** After analyzing a few offensive numbers I found that the conference with the highest average Offensive Efficiency is the Big 10 with . I also graphed the offensive against Wins and discovered that there is a positive correlation between offensive efficiency and wins. In addition to Offensive efficiency I discovered that the conference with the best three point perctage is the Summit Conference with 35.75%

#Defensive Statistic Analysis
**What I did:** The second thing I wanted to analyze is the effect defense has on wininng. To to this I graphed the defense efficiency to wins as well as graphed a bar chart of three point percentage.

In [15]:
df.head()

Team Conference  Games  ...  Adjusted Tempo  Wins Above Bubble  Seed
0  Michigan        B10     24  ...            66.9                7.2   1.0
1    Baylor        B12     24  ...            68.8                6.6   1.0
2  Illinois        B10     29  ...            70.7                8.9   1.0
3   Gonzaga        WCC     26  ...            74.6                8.5   1.0
4      Iowa        B10     29  ...            70.0                5.6   2.0

[5 rows x 22 columns]

In [16]:
df_average_defensive_stats = df.groupby('Conference', as_index=False).agg({
    'Wins':'mean',
    'Adj Defensive Efficiency':'mean',
    'Effective Field Goal Percentage Allowed':'mean',
    'Defensive Rebounds':'mean',
    'Two Point Percentage Allowed':'mean',
    'Three Point Percentage Allowed':'mean'})
df_average_defensive_stats = df_average_defensive_stats.sort_values('Wins', ascending = False)
df_average_defensive_stats

Conference  ...  Three Point Percentage Allowed
5         B10  ...                       33.221429
6         B12  ...                       33.620000
21        P12  ...                       34.300000
25        SEC  ...                       32.664286
24         SC  ...                       33.640000
17        MVC  ...                       34.210000
12       CUSA  ...                       34.014286
23         SB  ...                       34.850000
18        MWC  ...                       33.845455
20        OVC  ...                       34.441667
7          BE  ...                       33.254545
1         ACC  ...                       34.600000
30        WCC  ...                       33.770000
3        ASun  ...                       34.044444
4        Amer  ...                       32.500000
8        BSky  ...                       35.000000
15        MAC  ...                       33.783333
27       Slnd  ...                       34.246154
13       Horz  ...                       34.550000
9        BSth  ...                       33.745455
10         BW  ...                       35.154545
0         A10  ...                       32.742857
28        Sum  ...                       36.922222
29        WAC  ...                       34.122222
14       MAAC  ...                       32.627273
19        NEC  ...                       33.090000
11        CAA  ...                       33.560000
2          AE  ...                       32.770000
26       SWAC  ...                       33.300000
22        Pat  ...                       33.780000
16       MEAC  ...                       32.988889

[31 rows x 7 columns]

In [17]:
fig = px.bar(
    df_average_defensive_stats,
    x = 'Conference',
    y ='Adj Defensive Efficiency',
    title = 'Defensive Efficiency by Conference',
    color = 'Conference'
)
fig.show()

In [18]:
px.scatter(
    df_average_defensive_stats,
    x = 'Adj Defensive Efficiency',
    y = 'Wins',
    title = 'Is there a correlatuon between Defense Efficiency and Wins?',
    text = 'Conference',
    color = 'Conference'
)

In [19]:
fig = px.bar(
    df_average_defensive_stats,
    x = 'Conference',
    y = 'Three Point Percentage Allowed',
    title = 'Conference With The Best Three Point Percentage Defense',
    color = 'Conference',
    
)
fig.show()

**Findings:** After analyzing a few defensive metrics I discovered that the conference with the highest defensive efficiency is the Mid-eastern Athletic Conference and the conference with the lowest defensive efficiency is the Big 10. I also discovered by graphing wins to defensive efficiency that there is a negative correlation between wins and defensive efficiency. I also learned that the conference with the best three point defense is the SEC.

#Average Seed Analysis
**What I did:** The third thing I wanted to analyze was the tournament results of the conference. I used sorting and group by to help me analyze the tournement seeding of each conference respectively. I also decided to filter out all of the conferences with only 1 team in the tournement because those conferences are gaurenteed 1 team who is the conference winner

In [20]:
#Average Seed of conferences
df_average_conference_seed = df.groupby('Conference', as_index=False).agg({'Seed':['count','mean','median','max','min']})
df_average_conference_seed.columns = ['Conference','Total Seeds', 'Seed Mean', 'Seed Median', 'Seed Max', 'Seed Min']
df_average_conference_seed.sort_values(['Total Seeds', 'Seed Min'], ascending=[False,True], inplace=True)
df_average_conference_seed = df_average_conference_seed[df_average_conference_seed['Total Seeds'] > 1]
df_average_conference_seed

Conference  Total Seeds  Seed Mean  Seed Median  Seed Max  Seed Min
5         B10            9   5.555556          4.0      11.0       1.0
6         B12            7   4.000000          3.0       8.0       1.0
1         ACC            7   7.571429          8.0      11.0       4.0
25        SEC            6   5.666667          6.0       9.0       2.0
21        P12            5   8.200000          7.0      12.0       5.0
7          BE            4   7.250000          6.0      12.0       5.0
30        WCC            2   3.500000          3.5       6.0       1.0
4        Amer            2   6.500000          6.5      11.0       2.0
18        MWC            2   8.500000          8.5      11.0       6.0
17        MVC            2   9.500000          9.5      11.0       8.0
0         A10            2   9.500000          9.5      10.0       9.0

In [21]:
fig = px.box(df_average_conference_seed, x = 'Seed Mean', title = 'Average Seed Box Plot of Conferences with more then one Team')
fig.show()

In [22]:
fig = px.bar(df_average_conference_seed, x = 'Conference', y = 'Total Seeds', title = 'Total Seeds by Conference With More Then One Bid', color = 'Conference')
fig.show()

**Findings:**when analyzing how conferences were seeded in the tournement I excluded conferences with only 1 team to have a better understanding of of how the teams are actually seeded. If I had included conferences with 1 team it would be confusing since there would be several conferences with only 1. With that being said I learned that the Big Ten had the most teams in the tournement with 9. I also discovered that the average seed from a multiple bid conference was about 7 according to the box plot. The data was interesting to interpret becuase the averages were somewhat misleading. This was becuase several conferences had large ranges compared to others. For example, the Big Ten had a range from 1 to 11 with 9 data points whereas the WCC had a range from 1 to 6 with 2 data points. Therefore, to decide who was the best seeded conference I decided to focus on the median which led me to believe that the best seeded conference was the Big Ten

#Big Ten Analysis
**What I did:** In this section I compared key stats between Big Ten teams to see who had the best three point percentage, defense efficiency, offense efficiency, and wins. I ald created a scatter plot to see if there was a correlation between wins and tournement seed

In [23]:
df_B10 = df[df['Conference'] == 'B10']
df_B10.sort_values('Wins', ascending=False,inplace=True)
df_B10

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Team Conference  Games  ...  Adjusted Tempo  Wins Above Bubble  Seed
2        Illinois        B10     29  ...            70.7                8.9   1.0
4            Iowa        B10     29  ...            70.0                5.6   2.0
5        Ohio St.        B10     30  ...            66.9                6.0   2.0
0        Michigan        B10     24  ...            66.9                7.2   1.0
14         Purdue        B10     27  ...            66.6                3.8   4.0
34      Wisconsin        B10     29  ...            65.0                2.2   9.0
38       Maryland        B10     28  ...            65.4                0.2  10.0
36        Rutgers        B10     26  ...            67.8                1.9  10.0
40   Michigan St.        B10     27  ...            68.7                1.4  11.0
289     Minnesota        B10     29  ...            70.5               -1.1   NaN
247       Indiana        B10     27  ...            66.3               -1.2   NaN
217      Penn St.        B10     25  ...            69.1               -0.5   NaN
171  Northwestern        B10     23  ...            68.6               -2.9   NaN
138      Nebraska        B10     26  ...            72.1               -6.9   NaN

[14 rows x 22 columns]

In [24]:
fig = px.bar(
    df_B10,
    x = 'Team',
    y = 'Wins',
    title = 'Team Wins',
    color = 'Team'
)
fig.show()

In [25]:
fig = px.bar(
    df_B10,
    x = 'Team',
    y = 'Adj Defensive Efficiency',
    title = 'Defensive Efficiency by Team',
    color = 'Team'
)
fig.show()

In [26]:
fig = px.bar(
    df_B10,
    x = 'Team',
    y = 'Adj Offensive Efficiency',
    title = 'Offensive Efficiency by Team',
    color = 'Team'
)
fig.show()

In [27]:
fig = px.bar(
    df_B10,
    x = 'Team',
    y = 'Three Point Percentage',
    title = 'Team Three Point Percentage',
    color = 'Team'
)
fig.show()

In [28]:
fig =px.scatter(
    df_B10,
    x = 'Seed',
    y = 'Wins',
    title = 'Big Ten Seed Graph',
    color = 'Team',
    size = 'Wins'
)
fig.show()

**Findings:**I decided to analyze the Big Ten more in depth because I am an Illinois Fan and was curious how the Big Ten actually looked. After looking at the data I discovered that Ohio State had the highest defensive efficiency and that Iowa had the highest offensive efficiency. I was also curious about which team had the best three point cheating and after creating a graph I learned that Michigan barely took this one over Indiana with a three point shooting percentage of 38.7%

#**Conclusion**
This project was a lot of fun and enjoyed using what I have learned throughout the semester on a topic of strong interest to me. I am a huge college basketball fan so it was fun to create graphs and analyze college basketball data. It was interesting to learn that defensive efficiency has a negative corralation to wins since I have always heard that defense wins championships. I was not suprised by the postive correlation between wins and offensive efficiency. I also found it interesting taking a deeper dive into the seeding of major conferences. I was suprised how good the Big Ten and Big Twelve were. Both conferences had the most teams in the tournement and had a good median. 

In addition to this Jupyter Notebook I also performed several different analysises in Tableau. In Tableu I made several graphs to determine which team was best. I discoverd that Illinois had the most wins and best Two point percentage and that Michigan had the best three point percentage. I also graphed the the average three point percentage by conference and discovered that the Summit League had the best conference three point percentage which shocked me as I know hardly any teams in the conference. I also confirmed my python coding and learned that there is a postive correlation between offensive efficiency and wins and a negative correlation between defense efficiency and wins. I simplified my graph to only include tournement teams but it still confirmed my graphs from python. I also discovered in Tableau that Baylor had the best three point percentage as a team. I also discovered that multiple lower conference teams made the top of three point list after graphing only power five conference. I also graphed the average wins of every conference and learned that the Big Ten on average won more games.

Ultimatly I had a fun time playing around and expereimenting with all of the features of both Python and Tableau and after my analysis I can make the conclusion that the Big Ten is the best conference and that Illinois is the best team. My unbiased opinion of course ;). Overall this was a fun project and I really enjoyed the opprutunity to play around, experiment and learn.